src/index.js

```diff
@@ -1,47 +1,83 @@
 const listContainer = document.querySelector('.list');
+const addButton = document.querySelector('.add-btn');
+const clearAll = document.querySelector('#clear-btn');
 
-const task = [
-  {
-    description: 'task1',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'task2',
-    completed: false,
-    index: 2,
-  },
-  {
-    description: 'task3',
-    completed: false,
-    index: 3,
-  },
-  {
-    description: 'task4',
-    completed: false,
-    index: 4,
-  },
-];
-function iterateTasks() {
-  task.forEach((item) => {
-    listElement.innerHTML
-        += `
-        <div class="list-properties">
-            <span class="left-items">
-                <input type="checkbox">
-                <p>${item.description}</p>
-            </span>
-            <i class="fa-solid fa-ellipsis-vertical"></i>
-        </div>
-        <hr>
-        `;
+const displayToDoList = () => {
+  listContainer.innerHTML = '';
+  todoList.forEach((item, index) => {
+    const listItem = document.createElement('div');
+    listItem.className = 'list-properties';
+    listItem.innerHTML = `
+    <div class="both-sides">
+      <span class="left-items">
+        <input type="checkbox" ${item.completed ? 'checked' : ''}>
+        <p>${item.description}</p>
+      </span>
+      <span class="right-items">
+        <div id="remove-single"><i class="fa-solid fa-trash"></i></div>
+        <i class="fa-solid fa-ellipsis-vertical"></i>
+      </span>
+      </div>
+    `;
+    listItem.querySelector('#remove-single').addEventListener('click', (event) => {
+      event.preventDefault();
+      // eslint-disable-next-line no-use-before-define
+      removeItem(index);
+    });
+    listItem.querySelector('input[type="checkbox"]').addEventListener('change', (event) => {
+      event.preventDefault();
+      // eslint-disable-next-line no-use-before-define
+      markCompleted(index);
+    });
+    listContainer.appendChild(listItem);
   });
+};
 
-  listContainer.appendChild(listElement);
-}
-iterateTasks();
\ No newline at end of file
+const saveTodoList = () => {
+  localStorage.setItem('todoList', JSON.stringify(todoList));
+};
+const addItem = () => {
+  const task = inputItem.value;
+  if (task) {
+    const taskItem = {
+      description: task,
+      completed: false,
+    };
+    todoList.push(taskItem);
+    inputItem.value = '';
+    displayToDoList();
+    saveTodoList();
+  }
+};
+
+const markCompleted = (itemIndex) => {
+  if (itemIndex >= 0 && itemIndex < todoList.length) {
+    todoList[itemIndex].completed = true;
+    displayToDoList();
+    saveTodoList();
+  }
+};
+
+// Remove items
+
+const removeItem = (itemIndex) => {
+  if (itemIndex >= 0 && itemIndex < todoList.length) {
+    todoList.splice(itemIndex, 1);
+    displayToDoList();
+    saveTodoList();
+  }
+};
+
+// Removed all completed tasks
+const removeCompletedItems = () => {
+  todoList = todoList.filter((item) => !item.completed);
+  displayToDoList();
+  saveTodoList();
+};
+
```

Comment:
- [ ] As suggested by the previous reviewer, kindly make the functionality to update an existing todo 👍 to follow the project requirements 😊

**Steps to implement the Todo Update Method ⬇️**

1. Get the localStorage Data
2. Find the specific todo with index
3. replace the existing value with the new value
4. Set the new array to your localStorage
5. Render your data with displaytodo method